modules/addItemform.ts

```diff
@@ -85,18 +65,58 @@ addItemForm.onsubmit = async (event) => {
   const price = 0; // TODO PRICE
   const section = sectionSelect.value;
 
-  await itemManager.addItem(
-    name,
-    Number(quantity),
-    quantityUnit,
-    Number(price),
-    section
-  );
+  if (isEditing && currentItemId) {
+    // in edit mode, update item
+    await itemManager.editItem({
+      id: currentItemId,
+      name,
+      quantity: Number(quantity),
+      quantityUnit,
+      price: Number(price),
+      section,
+    });
+    isEditing = false;
+    currentItemId = null;
+    // close the modal
+    // addItemModalManager?.hideModal();
+  } else {
+    // add item
+    await itemManager.addItem(
+      name,
+      Number(quantity),
+      quantityUnit,
+      Number(price),
+      section
+    );
+  }
 
   addItemForm?.reset();
   nameInput?.focus();
 };
 
+// helper to pre-populate the form
+const populateForm = (item: Item) => {
+  if (!item) return;
+
+  nameInput.value = item.name || '';
+  quantityInput.value = String(item.quantity) || '1';
+  quantityUnitSelect.value = item.quantityUnit || '';
+  sectionSelect.value = item.section || '';
+
+  nameInput.focus();
+};
+
+// function to open the modal for editing
+export const openEditModal = (item: Item) => {
+  console.log('Open edit modal');
+  // addItemModalManager?.showModal();
+  // isEditing = true;
+  // if (!item.id) return;
+  // currentItemId = item.id;
+  // populateForm(item);
+};
```

Comment:
Problem in the PR draft body could also be linked to this. I'm passing the whole item to the function responsible for toggling the modal's visibility. This isn't a great pattern, and I should reconsider how this data gets flooded through the form.